src/Capgemini.PowerApps.PackageDeployerTemplate/PackageTemplateBase.cs

```diff
@@ -287,19 +286,43 @@ public override bool AfterPrimaryImport()
         {
             this.ExecuteLifecycleEvent(nameof(this.AfterPrimaryImport), () =>
             {
-                if (this.ConfigDataStorage.ActivateDeactivateSLAs)
+                if (this.TemplateConfig.ActivateDeactivateSLAs)
                 {
                     this.SlaDeploymentService.ActivateAll();
-                    this.SlaDeploymentService.SetDefaultSlas(this.ConfigDataStorage.DefaultSlas);
+                    this.SlaDeploymentService.SetDefaultSlas(this.TemplateConfig.DefaultSlas.Select(sla => sla.Name));
                 }
 
-                this.ProcessDeploymentService.Deactivate(this.ConfigDataStorage.ProcessesToDeactivate);
-                this.SdkStepDeploymentService.Deactivate(this.ConfigDataStorage.SdkStepsToDeactivate);
-                this.DataImporterService.Import(this.ConfigDataStorage.DataImports?.Where(c => !c.ImportBeforeSolutions), this.PackageFolderPath);
-                this.ProcessDeploymentService.Activate(this.ConfigDataStorage.ProcessesToActivate);
-                this.WordTemplateImporterService.ImportWordTemplates(this.ConfigDataStorage.WordTemplates, this.PackageFolderPath);
-                this.FlowDeploymentService.ConnectConnectionReferences(this.ConnectionReferenceMappings);
-                this.FlowDeploymentService.ActivateFlows(this.ProcessedSolutions, this.ConfigDataStorage.FlowsToDeactivate);
+                this.ConnectionReferenceSvc.ConnectConnectionReferences(this.ConnectionReferenceMappings);
+
+                this.ProcessDeploymentService.SetStatesBySolution(
+                    this.ProcessedSolutions,
+                    this.TemplateConfig.ProcessesToDeactivate.Select(p => p.Name));
```

Comment:
This needs to be after the data import (otherwise processes that have dependencies on reference data won't be able to be activated)